src/software/ai/hl/stp/play/enemy_ball_placement_play_test.cpp

```diff
@@ -0,0 +1,150 @@
+#include "software/time/duration.h"
+#include "software/world/world.h"
+
+class EnemyBallPlacementPlayTest : public SimulatedPlayTestFixture
+{
+   protected:
+    Field field = Field::createSSLDivisionBField();
+};
+
+TEST_F(EnemyBallPlacementPlayTest, test_ball_placement_center)
+{
+    Point ball_placement_point(1, 0);
+    BallState ball_state(Point(0, 0), Vector(0, 0));
+    auto friendly_robots = TestUtil::createStationaryRobotStatesWithId(
+        {Point(0, -1.5), Point(-1, 0), Point(0, 1.5), Point(-1.5, -1.5), Point(-1.5, 0),
+         Point(-1.5, 1.5)});
+    setFriendlyGoalie(0);
+    auto enemy_robots = TestUtil::createStationaryRobotStatesWithId(
+        {Point(1, 0), Point(1, 2.5), Point(1, -2.5), field.enemyGoalCenter(),
+         field.enemyDefenseArea().negXNegYCorner(),
+         field.enemyDefenseArea().negXPosYCorner()});
+    setEnemyGoalie(0);
+    setAIPlay(TYPENAME(EnemyBallPlacementPlay));
+    GameState game_state;
+    game_state.updateRefereeCommand(RefereeCommand::BALL_PLACEMENT_THEM);
+    game_state.setBallPlacementPoint(ball_placement_point);
+    setGameState(game_state);
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
+            while (world_ptr->getMostRecentTimestamp() < Timestamp::fromSeconds(9.5))
+            {
+                yield("Timestamp not at 9.5s");
+            }
+        }};
+
+    std::vector<ValidationFunction> non_terminating_validation_functions = {};
+
+    runTest(field, ball_state, friendly_robots, enemy_robots,
+            terminating_validation_functions, non_terminating_validation_functions,
+            Duration::fromSeconds(10));
+}
+
+TEST_F(EnemyBallPlacementPlayTest, test_ball_placement_diagonal)
+{
+    Point ball_placement_point(1, 0);
+    BallState ball_state(Point(1.5, 2), Vector(0, 0));
+    auto friendly_robots = TestUtil::createStationaryRobotStatesWithId(
+        {Point(0, -1.5), Point(-1, 0), Point(0, 1.5), Point(-1.5, -1.5), Point(-1.5, 0),
+         Point(-1.5, 1.5)});
+    setFriendlyGoalie(0);
+    auto enemy_robots = TestUtil::createStationaryRobotStatesWithId(
+        {Point(1, 0), Point(1, 2.5), Point(1, -2.5), field.enemyGoalCenter(),
+         field.enemyDefenseArea().negXNegYCorner(),
+         field.enemyDefenseArea().negXPosYCorner()});
+    setEnemyGoalie(0);
+    setAIPlay(TYPENAME(EnemyBallPlacementPlay));
+    GameState game_state;
+    game_state.updateRefereeCommand(RefereeCommand::BALL_PLACEMENT_THEM);
+    game_state.setBallPlacementPoint(ball_placement_point);
+    setGameState(game_state);
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
+            while (world_ptr->getMostRecentTimestamp() < Timestamp::fromSeconds(9.5))
+            {
+                yield("Timestamp not at 9.5s");
+            }
```

Comment:
We need proper validation rather than waiting for 9.5s. There are other cases of this in this file